src/sqswatcher/plugins/slurm.py

```diff
@@ -61,55 +64,120 @@ def _reconfigure_nodes():
+    """
     # Get the current node list
-    node_list = _read_node_list()
+    node_list = _read_node_list(PCLUSTER_NODES_CONFIG)
     nodes_to_restart = []
     for event in update_events:
         if event.action == "REMOVE":
             node_name = "NodeName={0}".format(event.host.hostname)
             node_list = [node for node in node_list if node.split()[0] != node_name]
         elif event.action == "ADD":
-            # Add new node
-            new_node = "NodeName={nodename} CPUs={cpus} State=UNKNOWN\n".format(
-                nodename=event.host.hostname, cpus=event.host.slots
+            # Only include GPU info if instance has GPU
+            gpu_info = "Gres=gpu:tesla:{gpus} ".format(gpus=event.host.gpus) if event.host.gpus != 0 else ""
+            new_node = "NodeName={nodename} CPUs={cpus} {gpu_info}State=UNKNOWN\n".format(
+                nodename=event.host.hostname, cpus=event.host.slots, gpu_info=gpu_info
             )
+
             if new_node not in node_list:
                 node_list.append(new_node)
             # Restarting also if already in config cause it might have failed at the previous iteration
             nodes_to_restart.append(event.host.hostname)
 
-    _write_node_list(node_list, max_cluster_size, instance_properties)
+    return node_list, nodes_to_restart
+
+
+def _add_dummy_to_node_list(node_list, max_cluster_size, instance_properties):
+    dummy_nodes_count = max_cluster_size - len(node_list)
+    if dummy_nodes_count > 0:
+        # Include GPU information for dummy nodes if instance type has GPU
+        if instance_properties["gpus"] == 0:
+            gpu_info = ""
+        else:
+            gpu_info = "Gres=gpu:tesla:{0} ".format(instance_properties["gpus"])
+        node_list.insert(
+            0,
+            "NodeName=dummy-compute[1-{0}] CPUs={1} {2}State=FUTURE\n".format(
+                dummy_nodes_count, instance_properties["slots"], gpu_info
+            ),
+        )
+
+
+def _add_dummy_to_gres_node_list(gres_node_list, max_cluster_size, instance_properties):
+    dummy_nodes_count = max_cluster_size - len(gres_node_list)
+    # Only add dummy node in gres.conf if instance has GPU
+    if dummy_nodes_count > 0 and instance_properties["gpus"] != 0:
+        gres_node_list.insert(
+            0,
+            "NodeName=dummy-compute[1-{0}] Name=gpu Type=tesla File=/dev/nvidia[0-{1}]\n".format(
+                dummy_nodes_count, instance_properties["gpus"] - 1
+            ),
+        )
+
+
+def update_cluster(max_cluster_size, cluster_user, update_events, instance_properties):
+    """Update and write node lists( and gres_nodes if instance has GPU); restart relevant nodes."""
+    node_list, nodes_to_restart = _update_node_lists(update_events)
+    _add_dummy_to_node_list(node_list, max_cluster_size, instance_properties)
+    _write_node_list_to_file(node_list, PCLUSTER_NODES_CONFIG)
```

Comment:
it might be useful to have a log message to log that we are updating the file and print the list of nodes we are writing to the file.

Same for gres.

Reply:
added logging to `_write_node_list_to_file`